internal/cmd/server/start_alarm_subscription_server.go

```diff
@@ -220,12 +220,13 @@ func (c *AlarmSubscriptionServerCommand) run(cmd *cobra.Command, argv []string)
 	}
 
 	// Create the handler:
-	handler, err := service.NewAlarmSubscriptionHandler().
+	handler, err := service.NewSubscriptionHandler().
 		SetLogger(logger).
 		SetLoggingWrapper(loggingWrapper).
 		SetCloudID(cloudID).
 		SetExtensions(extensions...).
 		SetKubeClient(kubeClient).
+		SetSubscriptionType("ALARM").
```

Comment:
I wonder if the storage should be created outside of the handler and then passed as a parameter instead of forcing the subscription handler to know the type of subscription. Is there any difference in the behaviour other than the namespace/name of the configmap where the subscriptions are stored?

Reply:
I made the configuration map configurable in my current PR. but not committed.

Reply:
There are also 2 other instances where we need to translate the reponse. The Alarm and Infrastructure subscription responses differ a bit. In [decodeSubId](https://github.com/openshift-kni/oran-o2ims/pull/107/files#diff-8bac4a7cd1e160e980214c0ee0cad277404fb16c3891ddca485a411965c31502R484-R491) and [encodeSubId](https://github.com/openshift-kni/oran-o2ims/pull/107/files#diff-8bac4a7cd1e160e980214c0ee0cad277404fb16c3891ddca485a411965c31502R447-R471). In these cases the handler needs to know the subscription type so that it translates correctly.

Reply:
@Jennifer-chen-rh , that's great. Would it be ok if we tried to merge the current PR first? Then yours will actually be shorter. The rebase shouldn't be a headache.

Reply:
Sure. Mine has more changes and comparatively harder to be a rebase code base.